core/src/main/java/com/linecorp/armeria/client/retry/RetryRuleUtil.java

```diff
@@ -16,14 +16,154 @@
+                try (HttpResponseDuplicator duplicator = ((HttpResponse) response).toDuplicator()) {
+                    final RetryRuleWithContent<T> duplicatedSecond =
+                            (RetryRuleWithContent<T>) withDuplicator(
+                                    (RetryRuleWithContent<HttpResponse>) second, duplicator);
+                    return orElse(ctx, response, fromRetryRule(first), duplicatedSecond);
+                }
+            } else {
+                return orElse(ctx, response, fromRetryRule(first), second);
+            }
+        };
+    }
+
+    @SuppressWarnings("unchecked")
+    static <T extends Response> RetryRuleWithContent<T> orElse(RetryRuleWithContent<T> first,
+                                                               RetryRule second) {
+        return (ctx, response) -> {
+            if (response instanceof HttpResponse) {
+                try (HttpResponseDuplicator duplicator = ((HttpResponse) response).toDuplicator()) {
+                    final RetryRuleWithContent<T> duplicatedFirst =
+                            (RetryRuleWithContent<T>) withDuplicator(
+                                    (RetryRuleWithContent<HttpResponse>) first, duplicator);
+                    return orElse(ctx, response, duplicatedFirst, fromRetryRule(second));
+                }
+            } else {
+                return orElse(ctx, response, first, fromRetryRule(second));
+            }
+        };
+    }
+
+    @SuppressWarnings("unchecked")
+    static <T extends Response> RetryRuleWithContent<T> orElse(RetryRuleWithContent<T> first,
+                                                               RetryRuleWithContent<T> second) {
+        return (ctx, response) -> {
+            if (response instanceof HttpResponse) {
+                try (HttpResponseDuplicator duplicator = ((HttpResponse) response).toDuplicator()) {
+                    final RetryRuleWithContent<T> duplicatedFirst =
+                            (RetryRuleWithContent<T>) withDuplicator(
+                                    (RetryRuleWithContent<HttpResponse>) first, duplicator);
+                    final RetryRuleWithContent<T> duplicatedSecond =
+                            (RetryRuleWithContent<T>) withDuplicator(
+                                    (RetryRuleWithContent<HttpResponse>) second, duplicator);
+                    return orElse(ctx, response, duplicatedFirst, duplicatedSecond);
+                }
+            } else {
+                return orElse(ctx, response, first, second);
+            }
+        };
+    }
+
+    private static <T extends Response> CompletionStage<RetryRuleDecision> orElse(
+            ClientRequestContext ctx, T response,
+            RetryRuleWithContent<T> first, RetryRuleWithContent<T> second) {
+        return orElse0(ctx, response, first::shouldRetry, second::shouldRetry);
+    }
+
+    private static <T> CompletionStage<RetryRuleDecision> orElse0(
```

Comment:
I'd prefer `orElse` if there's a conflict.

Reply:
There is an ambiguous method call if change `orElse0` to `orElse`. Let me leave it as is.